seed/serializers/properties.py

```diff
@@ -171,8 +172,25 @@ class Meta:
         }
 
     def to_representation(self, data):
-        """Overwritten to handle time conversion"""
+        """Overwritten to handle time conversion and extra_data null fields"""
         result = super(PropertyStateSerializer, self).to_representation(data)
+
+        if data.extra_data:
+            organization = data.organization
+            extra_data_columns = Column.objects.filter(
```

Comment:
I made progress in exploring this option a bit more, but I'm actually not sure which endpoints use this serializer. 

As a quick check, I added a `print` statement to the PropertyStateSerializer's `to_representation` method and hit the "127.0.0.1:8000/api/v2/properties/?organization_id=`<id>`" endpoint. I believe the `to_representation` method wasn't run as evidenced by not seeing output in the terminal where my server was running. Also, the output for that endpoint did not show pre-populated `extra_data` fields for returned properties.

I'm still learning the ins and outs of Django, so it's very possible that I'm missing something. I plan on investigating more, but could you easily point me to an endpoint that you know uses the PropertyStateSerializer so that I can check that the argument flag approach works as expected?

Reply:
I think adding an additional argument that passes in the `extra_data_columns` is probably the best. That way, the PropertyViewListSerializer can execute the query just once, instead of it being executed multiple times in the PropertyStateSerializer.

Yeah, I think you are seeing that there are more than one path through the Serializer world. The /api/v2/poperties goes through the PropertyViewSet (below), which does not explicitly call the PropertyStateSerializer (but should).

https://github.com/SEED-platform/seed/blob/f965e464f9c15d8e0e555c35f874304c9a8572ac/seed/views/properties.py#L203-L207